CI fails: ValueError: `num_hidden_layers` (2) must be equal to the number of layer types (64)

CI fails for Tests with dev dependencies: https://github.com/huggingface/trl/actions/runs/17769154310/job/50500273662
```python
ValueError: `num_hidden_layers` (2) must be equal to the number of layer types (64)
```
```python
=========================== short test summary info ============================
FAILED tests/test_grpo_trainer.py::GRPOTrainerTester::test_training_peft_with_gradient_checkpointing - ValueError: `num_hidden_layers` (2) must be equal to the number of layer types (64)
FAILED tests/test_grpo_trainer.py::GRPOTrainerTester::test_training_scale_rewards_0 - ValueError: `num_hidden_layers` (2) must be equal to the number of layer types (64)
FAILED tests/test_grpo_trainer.py::GRPOTrainerTester::test_training_scale_rewards_1_group - ValueError: `num_hidden_layers` (2) must be equal to the number of layer types (64)

...

FAILED tests/test_rloo_trainer.py::RLOOTrainerTester::test_training_peft - ValueError: `num_hidden_layers` (2) must be equal to the number of layer types (64)
FAILED tests/test_rloo_trainer.py::RLOOTrainerTester::test_training_peft_with_gradient_checkpointing - ValueError: `num_hidden_layers` (2) must be equal to the number of layer types (64)
= 71 failed, 840 passed, 46 skipped, 145 warnings, 356 rerun in 683.28s (0:11:23) =
```

I think this started after the merge in `transformers` of:
- huggingface/transformers#40791

with the addition of:
```diff
-         layer_type_validation(self.layer_types)
+         layer_type_validation(self.layer_types, self.num_hidden_layers)
```

Should we fix something on our side?

CC: @gante

Traceback:
```python
 _______ RLOOTrainerTester.test_training_peft_with_gradient_checkpointing _______
[gw1] linux -- Python 3.12.11 /__w/trl/trl/.venv/bin/python3

self = <tests.test_rloo_trainer.RLOOTrainerTester testMethod=test_training_peft_with_gradient_checkpointing>

    @require_peft
    def test_training_peft_with_gradient_checkpointing(self):
        """Test that training works with PEFT and gradient checkpointing enabled."""
        dataset = load_dataset("trl-internal-testing/zen", "standard_prompt_only", split="train")
    
        model = AutoModelForCausalLM.from_pretrained(
            "trl-internal-testing/tiny-Qwen2ForCausalLM-2.5",
            dtype=torch.float32,  # Use float32 for testing to avoid precision issues
        )
    
        lora_config = LoraConfig(
            r=8, lora_alpha=32, target_modules=["q_proj", "v_proj"], lora_dropout=0.05, bias="none"
        )
    
        training_args = RLOOConfig(
            output_dir=self.tmp_dir,
            learning_rate=0.1,
            per_device_train_batch_size=3,
            num_generations=3,
            max_completion_length=8,
            gradient_checkpointing=True,  # Enable gradient checkpointing
            report_to="none",
        )
>       trainer = RLOOTrainer(
            model=model,
            reward_funcs="trl-internal-testing/tiny-Qwen2ForSequenceClassification-2.5",
            args=training_args,
            train_dataset=dataset,
            peft_config=lora_config,
        )

tests/test_rloo_trainer.py:186: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
trl/trainer/rloo_trainer.py:359: in __init__
    reward_funcs[i] = AutoModelForSequenceClassification.from_pretrained(
.venv/lib/python3.12/site-packages/transformers/models/auto/auto_factory.py:549: in from_pretrained
    config, kwargs = AutoConfig.from_pretrained(
.venv/lib/python3.12/site-packages/transformers/models/auto/configuration_auto.py:1346: in from_pretrained
    return config_class.from_dict(config_dict, **unused_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
.venv/lib/python3.12/site-packages/transformers/configuration_utils.py:808: in from_dict
    config = cls(**config_dict)
             ^^^^^^^^^^^^^^^^^^
.venv/lib/python3.12/site-packages/transformers/models/qwen2/configuration_qwen2.py:210: in __init__
    layer_type_validation(self.layer_types, self.num_hidden_layers)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

layer_types = ['full_attention', 'full_attention', 'full_attention', 'full_attention', 'full_attention', 'full_attention', ...]
num_hidden_layers = 2

    def layer_type_validation(layer_types: list[str], num_hidden_layers: Optional[int] = None):
        """Check that `layer_types` is correctly defined."""
        if not all(layer_type in ALLOWED_LAYER_TYPES for layer_type in layer_types):
            raise ValueError(f"The `layer_types` entries must be in {ALLOWED_LAYER_TYPES}")
        if num_hidden_layers is not None and num_hidden_layers != len(layer_types):
>           raise ValueError(
                f"`num_hidden_layers` ({num_hidden_layers}) must be equal to the number of layer types "
                f"({len(layer_types)})"
            )
E           ValueError: `num_hidden_layers` (2) must be equal to the number of layer types (64)

.venv/lib/python3.12/site-packages/transformers/configuration_utils.py:1377: ValueError
```

--- a/scripts/generate_tiny_models.py
+++ b/scripts/generate_tiny_models.py
@@ -279,17 +279,21 @@ for model_id, model_class in [
     ("Qwen/Qwen2.5-VL-3B-Instruct", Qwen2_5_VLForConditionalGeneration),
 ]:
     processor = AutoProcessor.from_pretrained(model_id)
-    config = AutoConfig.from_pretrained(model_id)
 
-    config.text_config.num_hidden_layers = 2
-    config.text_config.hidden_size = 16
-    config.text_config.num_attention_heads = 4
-    config.text_config.num_key_value_heads = 2
-
-    config.vision_config.num_hidden_layers = 2
-    config.vision_config.hidden_size = 16
-    config.vision_config.num_attention_heads = 4
-    config.vision_config.num_key_value_heads = 2
+    text_config = {
+        "num_hidden_layers": 2,
+        "hidden_size": 16,
+        "num_attention_heads": 4,
+        "num_key_value_heads": 2,
+        "layer_types": None,  # Set it automatically from num_hidden_layers
+    }
+    vision_config = {
+        "num_hidden_layers": 2,
+        "hidden_size": 16,
+        "num_attention_heads": 4,
+        "num_key_value_heads": 2,
+    }
+    config = AutoConfig.from_pretrained(model_id, text_config=text_config, vision_config=vision_config)
 
     if isinstance(config, (Qwen2VLConfig)):
         config.vision_config.depth = 2

--- a/scripts/generate_tiny_models.py
+++ b/scripts/generate_tiny_models.py
@@ -300,6 +300,7 @@ for model_id, model_class in [
 
     if isinstance(config, (Qwen2VLConfig, Qwen2_5_VLConfig)):
         config.text_config.rope_scaling["mrope_section"] = [2]
+        config.rope_scaling["mrope_section"] = [2]  # different dict object from text_config; see GH-4101
 
     if isinstance(config, (Qwen2_5_VLConfig)):
         config.vision_config.out_hidden_size = 16

